lib/BoundaryValueDiffEqShooting/src/single_shooting.jl

```diff
@@ -34,25 +34,17 @@ function SciMLBase.__solve(prob::BVProblem, alg_::Shooting; odesolve_kwargs = (;
             u, p, ode_cache_loss_fn, bc, u0_size, prob.problem_type)
     end
 
-    sd = alg.jac_alg.diffmode isa AutoSparse ? SymbolicsSparsityDetection() :
-         NoSparsityDetection()
     y_ = similar(resid_prototype)
 
-    # Construct the jacobian function
-    # NOTE: We pass in a separate Jacobian Function because that allows us to cache the
-    #       the internal ode solve cache. This cache needs to be distinct from the regular
-    #       residual function cache
     jac_cache = if iip
-        sparse_jacobian_cache(alg.jac_alg.diffmode, sd, nothing, y_, vec(u0))
+        DI.prepare_jacobian(nothing, y_, diffmode, vec(u0))
     else
-        sparse_jacobian_cache(alg.jac_alg.diffmode, sd, nothing, vec(u0); fx = y_)
+        DI.prepare_jacobian(nothing, diffmode, vec(u0))
     end
```

Comment:
Preparation on `nothing` is not valid

Reply:
From testing, preparation on `nothing` is indeed valid, we prepare on `nothing` just to get the Dualized array in the preparation to construct the Jacobian of ODE-solving